pkg/native/serializers/serializer_cdx.go

```diff
@@ -45,6 +47,10 @@ func (s *CDX) Serialize(bom *sbom.Document, _ *native.SerializeOptions, _ interf
 
 	doc := cdx.NewBOM()
 	doc.SerialNumber = bom.Metadata.Id
+	if doc.SerialNumber == "" || !isValidCycloneDXSerialNumber(doc.SerialNumber) {
+		doc.SerialNumber = "urn:uuid:" + uuid.New().String()
```

Comment:
Since this is essentially altering the protobom data, we have an option to fail the conversion when the protobom document does not have an Id set. 

See how the SPDX serializer implements it here and make sure we honor it here too:
https://github.com/protobom/protobom/blob/5612e368b95d757401f322ed4939b3957d8d40a0/pkg/native/serializers/serializer_spdx23.go#L88-L95